File: 백준/Silver/15652. N과 M （4）/N과 M （4）.py
import sys

input = sys.stdin.readline

n, k = map(int, input().split())

rs = []
check = [False] * (n + 1)


def back(n, k1):
    if k1 == 0:
        print(" ".join(map(str, rs)))
        return

    for i in range(1, n + 1):
        if not rs or i >= rs[-1]:
            check[i] = True
            rs.append(i)
            back(n, k1 - 1)
            rs.pop()
            check[i] = False


back(n, k)

File: 백준/Silver/15652. N과 M （4）/N과 M （4）.js
const fs = require('fs');

const [n, k] = fs.readFileSync('/dev/stdin').toString().split(" ").map(Number);

const li = []
let answer = "";
const recur = () => {
    if(li.length === k){
        answer += li.join(" ") + "\n";
        return;
    }
    
    for (let i = 1; i <= n; i++){
        if(li.length === 0 || i >= li.at(-1)) {
            li.push(i)
            recur()
            li.pop()
        }
    }
}

recur()
console.log(answer);